### DISCUSSION TITLE
[PSD12] Exercise 3.20

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#182] [Esercizi (A-L)]
**Exercise 3.20** You are designing an FSM to keep track of the mood of four students working in the digital design lab. Each student’s mood is either HAPPY (the circuit works), SAD (the circuit blew up), BUSY (working on the circuit), CLUELESS (confused about the circuit), or ASLEEP (face down on the circuit board). How many states does the FSM have? What is the minimum number of bits necessary to represent these states?

**Comment by Elia-Belli** [comment]
Ognuno tra i 4 studente può avere 5 mood, quindi abbiamo $5^4=625$ possibili disposizioni (con ripetizione) di mood.
Per codificarli abbiamo bisogno di $log_2\ 625 \simeq 10$ bit (approssimato per eccesso)

**Comment by fraxb04** [comment]
corretto
